src/Orleans/Versions/Placement/IPlacementVersionDirector.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Orleans.Versions.Compatibility;
+using System;
+using System.Collections.Generic;
+
+namespace Orleans.Versions.Placement
+{
+    public interface IVersionPlacementDirector
+    {
+        IReadOnlyList<ushort> GetSuitableVersion(ushort requestedVersion, IReadOnlyList<ushort> availableVersions, IVersionCompatibilityDirector versionCompatibilityDirector);
```

Comment:
This API does not allow for weighted decisions... (like instead of randomly placing in all suitable silos, favor either the higher version if there is a big enough deployment rollout of that new version).
I'm fine that we don't over engineer this until we actually provide some of those directors in-the-box, but since it's an extension point, just wanted to raise awareness so we make a conscious choice not to support those for now.